spinalcordtoolbox/compat/launcher.py

```diff
@@ -9,39 +9,47 @@
 import os
 import subprocess
 import multiprocessing
+import importlib
 
 from spinalcordtoolbox import __file__ as package_init_file
+from spinalcordtoolbox.utils.sys import init_sct
+from spinalcordtoolbox.utils.fs import generate_json_sidecar
 
 
-def main():
+def main(argv=None):
     """
     Compatibility entry point to run scripts
     """
-
-    # Force scripts to not use graphical output
-    env = dict()
-    env.update(os.environ)
+    if argv is None:
+        argv = sys.argv[1:]
 
     if "ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS" not in os.environ:
-        env["ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS"] = str(multiprocessing.cpu_count())
+        os.environ["ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS"] = str(multiprocessing.cpu_count())
 
     # Needed to allow `sct_check_dependencies` to import voxelmorph/neurite without
     # failing due to a missing `tensorflow` dependency (since the backend defaults to TF)
-    env['VXM_BACKEND'] = 'pytorch'
-    env['NEURITE_BACKEND'] = 'pytorch'
+    os.environ['VXM_BACKEND'] = 'pytorch'
+    os.environ['NEURITE_BACKEND'] = 'pytorch'
 
     command = os.path.basename(sys.argv[0])
     pkg_dir = os.path.dirname(package_init_file)
 
     script = os.path.join(pkg_dir, "scripts", "{}.py".format(command))
     assert os.path.exists(script)
 
-    cmd = [sys.executable, script] + sys.argv[1:]
-
     mpi_flags = os.environ.get("SCT_MPI_MODE", None)
     if mpi_flags is not None:
         if mpi_flags == "yes":  # compat
             mpi_flags = "-n 1"
-        cmd = ["mpiexec"] + mpi_flags.split() + cmd
-
-    return subprocess.run(cmd, env=env).returncode
+        cmd = ["mpiexec"] + mpi_flags.split() + [sys.executable, script] + argv
+        return subprocess.run(cmd, env=os.environ).returncode
```

Comment:
Unfortunately, this method doesn't generate a JSON sidecar when SCT is run in `MPI_MODE`, as a subprocess appears to be necessary to run the [`mpiexec`](https://www.mpich.org/static/docs/v3.4.x/www1/mpiexec.html) executable.

Maybe there's a way we can get around this? But perhaps it's just easier to ditch the `launcher.py` idea, and instead call `generate_json_sidecar` directly from each scritp.